### PR TITLE
feat(chore/test): implement visual testing

### PR DESCRIPTION
**Pull request by mohamedMok**
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [X] Yes
- [ ] No

## Does this PR introduce a breaking change?

read [What is a breaking change ?](https://mozaic.adeo.cloud/Contributing/Prerequisite/GitConventions/#breaking-changes-)

- [ ] Yes
- [X] No

## Describe the changes

Github issue number or Jira issue URL: 
https://design-system-adeo.atlassian.net/browse/MZC-182

### CODE REVIEW

**Comment by mohamedMok** [comment]
@remy115  Thanks for the review! I'm working on this fix! 
I should've put the PR in WIP. 
Thanks for testing 👍

File: packages/gatsby-theme-styleguide/src/templates/preview-page.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react'
```
Comment (gael-boyenval) [inline packages/gatsby-theme-styleguide/src/templates/preview-page.js]: remove

File: packages/gatsby-theme-styleguide/src/gatsby-components/Preview/PreviewFrame.js [hunk 2]
```diff
@@ -111,48 +119,15 @@ export class PreviewFrame extends PureComponent {
...
-    }
-
-    head.appendChild(style)
-
-    document.body.innerHTML =
-      preview.html !== undefined ? preview.html : 'No html'
-    document.body.style.margin = 0
-
-    this.setState({
-      iframeHeight: this.refs.iframe.contentDocument.body.offsetHeight,
```
Comment (gael-boyenval) [inline packages/gatsby-theme-styleguide/src/gatsby-components/Preview/PreviewFrame.js]: keep this for iframes heights

Comment (gael-boyenval) [reply packages/gatsby-theme-styleguide/src/gatsby-components/Preview/PreviewFrame.js]: when switching to full screen and playing with viewport and restitching to normal => bug of iframe heights
=> recalculate height each time we change viewport as well as full screen

**Comment by gael-boyenval** [comment]
@mohamedMok : there

File: packages/gatsby-theme-styleguide/src/gatsby-components/Preview/PreviewFrame.js [hunk 3]
```diff
@@ -111,48 +119,23 @@ export class PreviewFrame extends PureComponent {
...
-    document.body.innerHTML =
-      preview.html !== undefined ? preview.html : 'No html'
-    document.body.style.margin = 0
-
+  iframeLoaded = () => {
     this.setState({
-      iframeHeight: this.refs.iframe.contentDocument.body.offsetHeight,
-      r: this.props.r,
+      iframeHeight: this.refs.iframe.contentDocument.body
+        ? this.refs.iframe.contentDocument.body.scrollHeight
```
Comment (gael-boyenval) [inline packages/gatsby-theme-styleguide/src/gatsby-components/Preview/PreviewFrame.js]: there is a scroll bar in the preview

**Pull request closed without merging**